### DISCUSSION TITLE
Interested in how you implemented this

### DISCUSSION THREAD

**Discussion by trisct** [arogozhnikov/einops#102] [Q&A] [answered]
Thanks for the great work!

Sorry I don't have enough experience to figure this out from the source code by myself, but I'd like to know how you implemented einops. If an operation has to be done in the backend framework in a few steps but can be done with einops in one step, do you first figure out the backend methods needed and then call them sequentially or do you manipulate the tensor directly in a more low-level way?

**Comment by trisct** [comment]
Additionally, is there a difference in memory usage in the above case?

**Comment by arogozhnikov** [comment, marked as answer]
Hi @trisct,

`einops` figures out an optimal set and sequence of operations to be called in a backend and then calls operations that are implemented by backend. Operations are packed to use minimal amount of operations independently on the complexity of transformation.

> Additionally, is there a difference in memory usage in the above case?

There should be no overhead in memory

**Comment by arogozhnikov** [comment]
Moved issue to discussions
